blog/2024-04-22.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Reorganizing our docs
+authors: [gagdiez]
+slug: reorganizing-docs
+tags: [docusaurus, updates]
+hide_table_of_contents: true
+---
+
+*We released a mayor reorganization of our repository, so we can improve docs for everyone... including us*
+
+<p><img src="https://img.freepik.com/free-photo/ancient-books-adorn-library-carefully-arranged-with-classics-rare-gems_157027-2332.jpg" /></p>
+
+<!-- truncate -->
+
+## Organic growth
+Our documentation is the result of multiple people collaborating across the span of four very active years, and it has seen a lot of changes: [2942 commits and counting](https://github.com/near/docs/commits/master/).
+
+At the beginning our docs only needed to explain how to create [smart contracts](/build/smart-contracts/what-is), and how to [interact with them through a frontend](/build/web3-apps/quickstart). Fast forward to today, and we have more than 200 pages of documentation, covering topics such as [chain abstraction](/build/chain-abstraction/what-is), [on-chain components](/build/near-components/what-is), [data infrastructure](/build/data-infrastructure/what-is), and [primitives such as NFT, FT](/build/primitives/what-is).
+
+The best is that new features keep being released every single month. However, all progress comes at a cost, and as our ecosystem grew, so did the disorganization of our documentation.
+
+## What link was that again?
+Let's briefly explain how [docusaurus](https://docusaurus.io/) (the framework we use in our docs) works so you can understand the problem.
+
+In docusaurus, all the pages are written as a simple markdown file. These files go inside the `./docs` folder, and can be organized in folders. Each file has a unique ID on its header that identifies it (e.g. `id: what-is`), and this ID alongside its folder path is used to generate the URL. 
+
+> For example, the document [`docs/build/smart-contracts/what-is.md`](https://github.com/near/docs/blob/master/docs/2.build/2.smart-contracts/what-is.md) has the `id: what-is`, so it ends ends being served in the URL https://docs.near.org/build/smart-contracts/what-is.
+
+### The problem
+
+About a year ago we noticed that our organic growth had left us with a very inconsistent URL structure. Basically, we had a lot of folders, and the files related to a same topic (e.g. NEAR components) would be all over the place.
+
+For example, you would be in the "Build" section reading about "What is a NEAR Component?" and the URL was `/bos/tutorial/quickstart`. The next page was "Setup an Environment" located at was `/bos/dev/intro`, followed by "Anatomy of a Component -> State" at `/bos/api/state`. Talk about consistency!
+
+Of course, we did not do this on purpose, it is just how things evolved. You might even notice that we are now talking about "NEAR Components" but the URL talks about "BOS". This is because when we started "BOS" (Blockchain Operating System) felt like a good name, but community feedback made us know that indeed, it was not. 
```

Comment:
```suggestion
Of course, we did not do this on purpose, it is just how things evolved. You might even notice that we are now talking about "NEAR Components" but the URL talks about "BOS". This is because when we started, "BOS" (Blockchain Operating System) felt like a good name, but community feedback made us know that, indeed, it was not. 
```